includes/rest/class-nodeinfo-controller.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * NodeInfo controller file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Rest;
+
+use function Activitypub\get_total_users;
+use function Activitypub\get_active_users;
+use function Activitypub\get_rest_url_by_path;
+
+/**
+ * ActivityPub NodeInfo Controller.
+ *
+ * @author Matthias Pfefferle
+ *
+ * @see https://nodeinfo.diaspora.software/
+ */
+class Nodeinfo_Controller extends \WP_REST_Controller {
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The REST base for this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'nodeinfo';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<version>\d\.\d)',
+			array(
+				'args' => array(
+					'version' => array(
+						'description' => 'The version of the NodeInfo schema.',
+						'type'        => 'string',
+						'required'    => true,
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves the NodeInfo discovery profile.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 *
+	 * @return \WP_REST_Response Response object.
+	 */
+	public function get_items( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$response = array(
+			'links' => array(
+				array(
+					'rel'  => 'https://nodeinfo.diaspora.software/ns/schema/2.0',
+					'href' => get_rest_url_by_path( '/nodeinfo' ),
```

Comment:
The version is missing

```suggestion
					'href' => get_rest_url_by_path( '/nodeinfo/2.0' ),
```

Reply:
Nice catch! Fixed in 893a128.